object.c

```diff
@@ -3184,35 +3184,71 @@ rb_opts_exception_p(VALUE opts, int default_value)
- *  spaces are optional.  This behavior is different from that of
- *  String#to_i.  Non string values will be converted by first
- *  trying <code>to_int</code>, then <code>to_i</code>.
- *
- *  Passing <code>nil</code> raises a TypeError, while passing a String that
- *  does not conform with numeric representation raises an ArgumentError.
- *  This behavior can be altered by passing <code>exception: false</code>,
- *  in this case a not convertible value will return <code>nil</code>.
- *
- *     Integer(123.999)    #=> 123
- *     Integer("0x1a")     #=> 26
- *     Integer(Time.new)   #=> 1204973019
- *     Integer("0930", 10) #=> 930
- *     Integer("111", 2)   #=> 7
- *     Integer(" +1_0 ")   #=> 10
- *     Integer(nil)        #=> TypeError: can't convert nil into Integer
- *     Integer("x")        #=> ArgumentError: invalid value for Integer(): "x"
- *
- *     Integer("x", exception: false)        #=> nil
+ *    Integer(object, base = 0, exception: true) -> integer or nil
+ *
+ *  Returns an integer converted from +object+.
+ *
+ *  Tries to convert +object+ to an integer
+ *  using +to_int+ first and +to_i+ second;
+ *  exception: when +object+ is a string, does _not_ call String#to_i
+ *  (see below).
+ *
+ *  With integer argument +object+ given, returns +object+:
+ *
+ *    Integer(1)                # => 1
+ *    Integer(-1)               # => -1
+ *
+ *  With floating-point argument +object+ given,
+ *  returns +object+ truncated to an intger:
+ *
+ *    Integer(1.9)              # => 1  # Rounds toward zero.
+ *    Integer(-1.9)             # => -1 # Rounds toward zero.
+ *
+ *  With string argument +object+ and zero +base+ given,
+ *  returns +object+ converted to an integer in base 10:
+ *
+ *    Integer('100')    # => 100
+ *    Integer('-100')   # => -100
+ *
+ *  With +base+ zero, string +object+ may contain leading characters
+ *  to specify the actual base:
+ *
+ *    Integer('0100')  # => 64  # Leading '0' specifies base 8.
+ *    Integer('0b100') # => 4   # Leading '0b', specifies base 2.
+ *    Integer('0x100') # => 256 # Leading '0x' specifies base 16.
+ *
+ *  With a non-zero +base+ (in range 2..36) given
+ *  (in which case +object+ must be a string),
+ *  returns +object+ converted to an integer in the given base:
+ *
+ *    Integer('100', 2)   # => 4
+ *    Integer('100', 8)   # => 64
+ *    Integer('100', 16)  # => 256
+ *    Integer('-100', 16) # => -256
+ *
+ *  When converting strings, surrounding whitespace and embedded underscores
+ *  are allowed and ignored:
+ *
+ *    Integer(' 100 ')      # => 100
+ *    Integer('-1_0_0', 16) # => -256
+ *
+ *  Examples with +object+ of various other classes:
+ *
+ *    Integer(Rational(9, 10))  # => 0  # Rounds toward zero.
+ *    Integer(Rational(-9, 10)) # => 0  # Rounds toward zero.
+ *    Integer(Complex(2, 0))    # => 2  # Imaginary part must be zero.
+ *    Integer(Time.now)          # => 1650974042
```

Comment:
Let's limit this section to 3 examples at most.  This just documents `to_int`/`to_i` behavior for the related objects, arguably it doesn't need to be included at all.

Reply:
Done.